web/src/main/java/org/cbioportal/web/util/DataBinHelper.java

```diff
@@ -89,64 +90,68 @@ else if (lteMax == null || (ltMax != null && lteMax < ltMax)) {
         // Find a generous IQR. This is generous because if (values.length / 4) 
         // is not an int, then really you should average the two elements on either 
         // side to find q1 and q3.
-        Range<Double> interquartileRange = calcInterquartileRange(sortedValues);
-
-        Double q1 = interquartileRange.lowerEndpoint();
-        Double q3 = interquartileRange.upperEndpoint();
-        Double iqr = q3 - q1;
+        Range<BigDecimal> interquartileRange = calcInterquartileRange(sortedValues);
+
+        BigDecimal q1 = interquartileRange.lowerEndpoint();
+        BigDecimal q3 = interquartileRange.upperEndpoint();
+        BigDecimal iqr = q3.subtract(q1);
+        BigDecimal iqrOneAndHalf = iqr.multiply(new BigDecimal("1.5"));
+        BigDecimal q1LowerBoundry = q1.subtract(iqrOneAndHalf);
+        BigDecimal q3upperBoundry = q3.add(iqrOneAndHalf);
+        
 
         // Then find min and max values
-        Double maxValue;
-        Double minValue;
+        BigDecimal maxValue;
+        BigDecimal minValue;
 
-        if (sortedValues.get(0).equals(sortedValues.get(sortedValues.size() - 1))) {
+        if (sortedValues.get(0).compareTo(sortedValues.get(sortedValues.size() - 1)) == 0) {
             // if the first and last values are the same, no need to do any other calculation
             // we simply set min and max to the same value
             minValue = sortedValues.get(0);
             maxValue = minValue;
         }
-        else if (0.001 <= q3 && q3 < 1.0) {
+        else if (q3.compareTo(new BigDecimal("0.001")) != -1 && q3.compareTo(new BigDecimal("1")) == -1) {
             //maxValue = Number((q3 + iqr * 1.5).toFixed(3));
             //minValue = Number((q1 - iqr * 1.5).toFixed(3));
-            maxValue = (new BigDecimal(q3 + iqr * 1.5)).setScale(3, BigDecimal.ROUND_HALF_UP).doubleValue();
-            minValue = (new BigDecimal(q1 - iqr * 1.5)).setScale(3, BigDecimal.ROUND_HALF_UP).doubleValue();
+            maxValue = q3upperBoundry.setScale(3, BigDecimal.ROUND_HALF_UP);
+            minValue = q1LowerBoundry.setScale(3, BigDecimal.ROUND_HALF_UP);
         } 
-        else if (q3 < 0.001) {
+        else if (q3.doubleValue() < 0.001) {
```

Comment:
why do we need to use `doubleValue()` here instead of `compareTo`?

Reply:
Ah, it should be converted.